src/firmware/app/control/trajectory_planner.c

```diff
@@ -124,33 +122,38 @@ app_trajectory_planner_generateConstantArcLengthTrajectory(
         generation_status = INITIAL_VELOCITY_TOO_HIGH;
     }
 
-
     // Now that the velocity profile has been defined we can calculate the time profile of
     // the trajectory
-    app_trajectory_planner_generateTimeProfile(traj_elements, num_segments,
-                                               arc_segment_length, velocity_profile);
+    app_trajectory_planner_generatePositionTrajectoryTimeProfile(
+        traj_elements, num_segments, arc_segment_length, velocity_profile);
 
     // If the parameterization ended up being reversed, we need to flip all the values
     // back.
     // TODO: Remove the 'reverse parameterization' hack #1322
     if (reverse_parameterization == true)
     {
-        app_trajectory_planner_reverseTrajectoryDirection(traj_elements, num_segments);
+        app_trajectory_planner_reversePositionTrajectoryDirection(trajectory);
     }
 
     return generation_status;
 }
 
 
-void app_trajectory_planner_interpolateConstantTimeTrajectory(
-    Trajectory_t* constant_period_trajectory, Trajectory_t* variable_time_trajectory,
-    const float interpolation_period)
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_interpolateConstantTimePositionTrajectory(
+    PositionTrajectory_t* constant_period_trajectory,
```

Comment:
we use `ConstantTime` in the function name but `constant_period` for the argument. Could we change one or the other to be consistent? I'd lean towards `constant_period` because we also have the `interpolation_period` argument.

Reply:
Renamed to constant period